src/sagemaker/workflow/pipeline_context.py

```diff
@@ -122,9 +194,10 @@ def wrapper(*args, **kwargs):
                 UserWarning,
             )
             run_func(*args, **kwargs)
+            if run_func.__name__ in ["register", "create"]:
+                return _restructure_model_step_args(args[0].sagemaker_session.context)
```

Comment:
You can check the instance type of args[0] here.

Reply:
Sure, updated in the new commit